src/addrman.h

```diff
@@ -304,8 +336,14 @@ class CAddrMan
 
         unsigned char nVersion = 1;
         s << nVersion;
-        s << ((unsigned char)32);
+        s << ((unsigned char)36);
         s << nKey;
+        unsigned long long asmap_version = 0;
+        if (m_asmap.size() != 0) {
+            std::hash<std::vector<bool>> hasher;
+            asmap_version = hasher(m_asmap);
```

Comment:
nit: You might want to make it the same length in bits an all architectures:

```suggestion
            const uint64_t asmap_version{hasher(m_asmap)};
            s << asmap_version;
```